dll/win32/shell32/shlexec.cpp

```diff
@@ -802,6 +802,15 @@ static UINT SHELL_FindExecutable(LPCWSTR lpPath, LPCWSTR lpFile, LPCWSTR lpVerb,
         xlpFile[0] = '\0';
     }
 
+    /* Clear any trailing periods */
+    for (SIZE_T i = wcslen(xlpFile); i > 0; i--)
+    {
+        if (i > 0 && xlpFile[i - 1] == '.')
+            xlpFile[i - 1] = '\0';
+        else
+            break;
+    }
```

Comment:
```suggestion
    SIZE_T i = wcslen(xlpFile);
    while (i > 0 && xlpFile[i - 1] == '.')
    {
        xlpFile[i - 1] = '\0';     // or: xlpFile[--i] = '\0';   // and you could remove the --i; on the next line.
        --i;
    }
```

Reply:
Can you point me to any specific benefits of making these changes. If no, then I prefer to stay with my previous solution. Thanks.

Reply:
It makes evident that the intent is to really only loop _**while**_ there are periods (and you get the "defensive" `i > 0` test that is not duplicated twice).

Reply:
OK. Suggestion accepted. See what you think now. Can you approve this? Thanks.